### PR TITLE
Documenting bags, berries and graphics

### PR DESCRIPTION
**Pull request by thechurchofcage**
Bag graphics and berry graphics mostly documented
item_menu.c entirely documented
Bag structs documented and functions renamed to match formatting

### CODE REVIEW

File: include/item_menu_icons.h [hunk 1]
```diff
@@ -19,7 +19,13 @@ void sub_80D4FC8(u8 arg0);
 void sub_80D4FEC(u8 arg0);
 u8 CreateBerryTagSprite(u8 id, s16 x, s16 y);
 void FreeBerryTagSpritePalette(void);
-u8 sub_80D511C(u8 berryId, u8 x, u8 y, bool8 startAffine);
+u8 LoadSpinningBerryPicGfx(u8 berryId, u8 x, u8 y, bool8 startAffine);
 u8 CreateBerryFlavorCircleSprite(s16 x);
 
+#define BAG_GFX 100
```
Comment (huderlem) [inline include/item_menu_icons.h]: These 5 defines are graphics tags, so they should all have a `TAG_` prefix in their names.  (i.e. `TAG_ROTATING_BALL`, TAG_BERRY_PIC`, etc.)

File: include/item_menu_icons.h [hunk 2]
```diff
@@ -1,11 +1,11 @@
 #ifndef GUARD_ITEM_MENU_ICONS
 #define GUARD_ITEM_MENU_ICONS
 
-extern const struct CompressedSpriteSheet gUnknown_0857FB34;
-extern const struct CompressedSpriteSheet gUnknown_0857FB3C;
-extern const struct CompressedSpritePalette gUnknown_0857FB44;
-extern const struct CompressedSpriteSheet gUnknown_0857FDEC;
-extern const struct CompressedSpritePalette gUnknown_0857FDF4;
+extern const struct CompressedSpriteSheet gBagMaleTable;
```
Comment (huderlem) [inline include/item_menu_icons.h]: These should be named more like `gBagMaleSpriteSheet`, rather than the `Table` suffix.

File: src/item_menu.c [hunk 3]
```diff
@@ -185,8 +185,8 @@ static const struct BgTemplate sBgTemplates_ItemMenu[3] =
 static const struct ListMenuTemplate sItemListMenu =
 {
     .items = NULL,
-    .moveCursorFunc = bag_menu_change_item_callback,
-    .itemPrintFunc = sub_81AB520,
+    .moveCursorFunc = BagMenu_ChangeItemCallback,
```
Comment (huderlem) [inline src/item_menu.c]: Can we call this "move cursor" rather than "change item"?

File: src/item_menu.c [hunk 4]
```diff
@@ -513,28 +513,28 @@ void sub_81AAC70(void)
...
-            gUnknown_0203CE54->unk81B = 1;
-        gUnknown_0203CE54->mainCallback2 = 0;
-        gUnknown_0203CE54->unk81A = 0xFF;
-        gUnknown_0203CE54->unk81E = -1;
-        gUnknown_0203CE54->unk81F = -1;
-        memset(gUnknown_0203CE54->spriteId, 0xFF, sizeof(gUnknown_0203CE54->spriteId));
-        memset(gUnknown_0203CE54->windowPointers, 0xFF, 10);
+            gBagTaskStruct->unk81B = 1;
+        gBagTaskStruct->mainCallback2 = 0;
+        gBagTaskStruct->unk81A = 0xFF;
```
Comment (huderlem) [inline src/item_menu.c]: Looks like `gBagTaskStruct` is just supposed to be called `gItemMenu` or `gBagMenu`.

**Comment by thechurchofcage** [comment]
Made the changes you requested 👍

File: src/item_menu.c [hunk 5]
```diff
@@ -144,10 +144,10 @@ void item_menu_type_2(u8 taskId);
 void display_sell_item_ask_str(u8 taskId);
 void display_deposit_item_ask_str(u8 taskId);
 void item_menu_type_b(u8 taskId);
-void BagMenuActuallyToss(u8 taskId);
-void BagMenuCancelToss(u8 taskId);
-void sub_81AD84C(u8 taskId);
-void sub_81AD6FC(u8 taskId);
+void BagMenu_ActuallyToss(u8 taskId);
```
Comment (huderlem) [inline src/item_menu.c]: `Confirm` instead of `Actually` for these.

**Comment by huderlem** [comment]
The build is failing:
```
src/item_menu.c:1724: redefinition of `BagMenu_ConfirmToss'
src/item_menu.c:1681: `BagMenu_ConfirmToss' previously defined here
```

**Pull request merged**